wremnants/combine_theory_helper.py

```diff
@@ -983,36 +983,55 @@ def add_pdf_uncertainty(self, operation=None, scale=-1.0):
                     **tmp_pdf_args,
                 )
 
-    def add_pdf_alphas_variation(self, noi=False, scale=-1.0):
+    def add_pdf_alphas_variation(self, noi=False):
         pdf = self.datagroups.args_from_metadata("pdfs")[0]
         pdfInfo = theory_tools.pdf_info_map("ZmumuPostVFP", pdf)
         pdfName = pdfInfo["name"]
-        scale = (
-            scale
-            if scale != -1.0
-            else theory_tools.pdf_inflation_factor(pdfInfo, self.args.noi)
-        )
-        pdf_hist = pdfName
-        pdf_corr_hist = (
-            f"scetlib_dyturbo{pdf.upper().replace('AN3LO', 'an3lo')}VarsCorr"
-            if self.corr_hist_name == "scetlib_dyturboCorr"
-            else self.corr_hist_name.replace("Corr", "VarsCorr")
-        )
         symmetrize = "average" if noi else "quadratic"
-        asRange = pdfInfo["alphasRange"]
-        as_replace = (
-            [("as", "pdfAlphaS")] + [("0116", "Down"), ("0120", "Up")]
-            if asRange == "002"
-            else [("0117", "Down"), ("0119", "Up")]
-        )
-        if self.from_hels:
-            asname = "pdfAlphaSByHelicity"
-        else:
-            asname = (
-                f"{pdfName}alphaS{asRange}"
-                if not self.as_from_corr
-                else pdf_corr_hist.replace("Vars", "_pdfas")
+        as_range = pdfInfo["alphasRange"]
+
+        if self.as_from_corr:
+            pdf_corr_hist = (
+                f"scetlib_dyturbo{pdf.upper().replace('AN3LO', 'an3lo')}VarsCorr"
+                if self.corr_hist_name == "scetlib_dyturboCorr"
+                else self.corr_hist_name.replace("Corr", "VarsCorr")
             )
+            asname = pdf_corr_hist.replace("Vars", "_pdfas")
+            # alphaS from correction histograms only available for these sets,
+            # so fall back to CT18Z for other sets
+            if not ("MSHT20" in asname or "CT18Z" in asname or "MSHT20an3lo" in asname):
+                asname = "scetlib_dyturboCT18Z_pdfasCorr"
+                as_range = "002"
```

Comment:
Can this directly be read from the CT18Z resources? I.e. 
```
as_range = theory_tools.pdfMap["ct18z"]["alphasRange"]
```

Reply:
fixed. This was from Josh's fixes